.github/workflows/test-publish.yml

```diff
@@ -28,7 +28,9 @@ jobs:
 
       - name: Install dependencies
         working-directory: .
-        run: poetry install
+        run: |
+          poetry env use 3.9.9
```

Comment:
in the yml file we specify version `python-version: "3.9"` of the image to use but here we require explicit (up to the patch) version, can this lead to some failure?